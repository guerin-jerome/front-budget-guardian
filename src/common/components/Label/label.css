label {
  font-family: Arial, sans-serif;
  font-size: 16px;
  font-weight: bold;
  color: #333;
}
