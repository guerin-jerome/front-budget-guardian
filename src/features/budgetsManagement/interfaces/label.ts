/**
 * Informations globales
 */
export const BUDGETS_TOTAL_TITLE_SECTION = "Total des budgets :";

/**
 * Labels du formulaire d'ajout d'un budget
 */
export const BUDGET_NAME_LABEL = "Nom :";
export const BUDGET_TYPE_LABEL = "Type :";
export const BUDGET_AMOUNT_LABEL = "Montant :";

/**
 * Labels de la carte d'affichage des budgets
 */
export const BUDGET_CARD_NAME_LABEL = "Nom du budget :";
export const BUDGET_CARD_TYPE_LABEL = "Type du budget :";
export const BUDGET_CARD_AMOUNT_LABEL = "Montant du budget :";

/**
 * Texte affiché lorsque l'utilisateur n'a pas de budget
 */
export const NO_BUDGET_TEXT =
  "Vous n'avez pas encore de budgets, vous pouvez en créer un.";
