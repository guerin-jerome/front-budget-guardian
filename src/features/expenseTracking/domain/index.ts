import { buildBudgetsImpactedOptions } from "./buildBudgetsImpactedOptions";
import { buildLibelleCurrentValue } from "./buildLibelleCurrentValue";
import { buildLibelleTypeBudget } from "./buildLibelleTypeBudget";
import { buildSubmitTextButtons } from "./buildSubmitTextButtons";
import { calculateBudgetConsumedPercentage } from "./calculateBudgetConsumedPercentage";
import { calculateTotalRemaining } from "./calculateTotalRemaining";
import { resetCurrentBudgets } from "./resetCurrentBudgets";

export {
  buildBudgetsImpactedOptions,
  buildLibelleCurrentValue,
  buildLibelleTypeBudget,
  buildSubmitTextButtons,
  calculateBudgetConsumedPercentage,
  calculateTotalRemaining,
  resetCurrentBudgets,
};
