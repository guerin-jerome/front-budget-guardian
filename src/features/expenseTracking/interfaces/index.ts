import { ExpenseTracking } from "./ExpenseTracking";

export { ExpenseTracking };
