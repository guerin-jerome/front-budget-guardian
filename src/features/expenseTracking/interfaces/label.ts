export const EXPENSE_TRACKING_TITLE = "Suivi des dépenses";

/**
 * Section du formulaire d'ajout d'une dépense
 */
export const ADD_EXPENSE_TITLE_SECTION = "Ajouter une dépense";
export const EXPENSE_DETAILS_LABEL = "Détails";
export const EXPENSE_BUDGET_IMPACTED_LABEL = "Budget impacté";
export const EXPENSE_DATE_LABEL = "Date";
export const EXPENSE_AMOUNT_LABEL = "Montant";
export const ADD_EXPENSE_TEXT_MOBILE = "Ajouter";
export const ADD_EXPENSE_TEXT_DESKTOP = "+";
export const REMOVE_EXPENSE_TEXT_MOBILE = "Retirer";
export const REMOVE_EXPENSE_TEXT_DESKTOP = "-";

/**
 * Section de la liste des budgets en cours
 */
export const TOTAL_REMAINING_TITLE_SECTION = "Total restant : ";
export const RESET_BUTTON_TEXT = "Réinitialiser";
