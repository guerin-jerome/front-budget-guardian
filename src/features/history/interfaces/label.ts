export const DETAILS_INPUT_LABEL = "DÃ©tails :";
export const BUDGET_IMPACTED_SELECT_LABEL = "Budget :";
export const BUDGET_TYPE_SELECT_LABEL = "Type :";
export const SORT_ORDER_BUDGET_DATE_LABEL = "Trier par date :";
