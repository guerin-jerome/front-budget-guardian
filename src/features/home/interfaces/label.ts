export const MENU_ITEM_EXPENSES_TRACKING_TEXT_DESKTOP = "Suivi des dépenses";
export const MENU_ITEM_EXPENSES_TRACKING_TEXT_MOBILE = "Suivi";
export const MENU_ITEM_BUDGETS_MANAGEMENT_TEXT_DESKTOP = "Gestion des budgets";
export const MENU_ITEM_BUDGETS_MANAGEMENT_TEXT_MOBILE = "Gestion";
export const HISTORY_TITLE = "Historique des dépenses";
export const MENU_ITEM_HISTORIQUE_TEXT = "Historique";
