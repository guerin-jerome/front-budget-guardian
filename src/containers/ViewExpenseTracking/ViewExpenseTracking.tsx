import { ViewCurrentBudget } from "./ViewCurrentBudget/ViewCurrentBudget";
import { Typography } from "../../components/Typography/Typography";
import { useExpenseTracking } from "../../hooks/useExpenseTracking";
import { EXPENSE_TRACKING_TITLE } from "./expenseTracking.label";
import "./expenseTracking.css";

const { Title, Text } = Typography;

export const ViewExpenseTracking = () => {
  const { budgets, totalRemaining } = useExpenseTracking();

  return (
    <div className="container">
      <Title level={2}>{EXPENSE_TRACKING_TITLE}</Title>
      <Text>Total restant : {totalRemaining} â‚¬</Text>
      {budgets.map((budget) => (
        <ViewCurrentBudget {...budget} key={budget.id} />
      ))}
    </div>
  );
};
