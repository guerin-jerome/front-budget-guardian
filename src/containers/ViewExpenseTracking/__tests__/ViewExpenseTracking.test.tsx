import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { ViewExpenseTracking } from "../ViewExpenseTracking";
import { EXPENSE_TRACKING_TITLE } from "../expenseTracking.label";

describe("<ViewCurrentBudget />", () => {
  it("matchs snapshot", () => {
    const { container } = render(<ViewExpenseTracking />);
    expect(container).toMatchSnapshot();
  });

  it("renders well", () => {
    render(<ViewExpenseTracking />);
    const expenseTrackingTitle = screen.getByRole("heading", {
      name: EXPENSE_TRACKING_TITLE,
    });
    const totalRemainingText = screen.getByText("Total restant : 219,4 â‚¬");
    expect(expenseTrackingTitle).toBeDefined();
    expect(totalRemainingText).toBeDefined();
  });
});
