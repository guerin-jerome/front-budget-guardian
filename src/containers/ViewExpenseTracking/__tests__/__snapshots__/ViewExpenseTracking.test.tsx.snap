// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ViewCurrentBudget /> > matchs snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <h2>
      Suivi des dépenses
    </h2>
    <p>
      Total restant : 
      219,4
       €
    </p>
    <div
      class="budget"
    >
      <h3>
        Loisirs
      </h3>
      <div
        class="budget-info"
      >
        <span
          class="budget-type"
        >
          Dépense variable
        </span>
        <span
          class="budget-value"
        >
          5,6€/100€
        </span>
      </div>
      <div
        class="progress-bar"
      >
        <div
          class="progress"
          style="width: 5%;"
        />
      </div>
    </div>
    <div
      class="budget"
    >
      <h3>
        Placement financier
      </h3>
      <div
        class="budget-info"
      >
        <span
          class="budget-type"
        >
          Épargne
        </span>
        <span
          class="budget-value"
        >
          200€
        </span>
      </div>
      <div
        class="progress-bar"
      >
        <div
          class="progress"
          style="width: 100%;"
        />
      </div>
    </div>
    <div
      class="budget"
    >
      <h3>
        Loyer
      </h3>
      <div
        class="budget-info"
      >
        <span
          class="budget-type"
        >
          Dépense fixe
        </span>
        <span
          class="budget-value"
        >
          560€/560€
        </span>
      </div>
      <div
        class="progress-bar"
      >
        <div
          class="progress"
          style="width: 100%;"
        />
      </div>
    </div>
    <div
      class="budget"
    >
      <h3>
        Courses
      </h3>
      <div
        class="budget-info"
      >
        <span
          class="budget-type"
        >
          Dépense variable
        </span>
        <span
          class="budget-value"
        >
          50€/175€
        </span>
      </div>
      <div
        class="progress-bar"
      >
        <div
          class="progress"
          style="width: 28%;"
        />
      </div>
    </div>
  </div>
</div>
`;
